pkg/csi/rbddaemonset.go

```diff
@@ -52,349 +51,295 @@ var (
+						},
+					},
+				},
+				{
+					Name: "registration-dir",
+					VolumeSource: corev1.VolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: fmt.Sprintf("%s/plugins_registry/", templates.DefaultKubeletDirPath),
+							Type: &hostPathDirectory,
+						},
+					},
+				},
+				{
+					Name: "pods-mount-dir",
+					VolumeSource: corev1.VolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: fmt.Sprintf("%s/pods", templates.DefaultKubeletDirPath),
+							Type: &hostPathDirectory,
+						},
+					},
+				},
+				{
+					Name: "ceph-csi-kms-config",
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: templates.EncryptionConfigMapName,
+							},
+							Items: []corev1.KeyToPath{
+								{
+									Key:  "config.json",
+									Path: "config.json",
 								},
 							},
 						},
-						{
-							Name: "oidc-token", VolumeSource: corev1.VolumeSource{
-								Projected: &corev1.ProjectedVolumeSource{
-									Sources: []corev1.VolumeProjection{
-										{
-											ServiceAccountToken: &corev1.ServiceAccountTokenProjection{
-												Path:              "oidc-token",
-												ExpirationSeconds: ptr.To(int64(3600)),
-												Audience:          "ceph-csi-kms",
-											},
-										},
+					},
+				},
+				{
+					Name: "oidc-token", VolumeSource: corev1.VolumeSource{
+						Projected: &corev1.ProjectedVolumeSource{
+							Sources: []corev1.VolumeProjection{
+								{
+									ServiceAccountToken: &corev1.ServiceAccountTokenProjection{
+										Path:              "oidc-token",
+										ExpirationSeconds: ptr.To(int64(3600)),
+										Audience:          "ceph-csi-kms",
 									},
 								},
 							},
 						},
 					},
-					Tolerations: []corev1.Toleration{
-						utils.GetTolerationForCSIPods(),
-					},
 				},
 			},
+			Tolerations: []corev1.Toleration{
+				utils.GetTolerationForCSIPods(),
+			},
 		},
+	},
+}
+
+func SetRBDDaemonSetDesiredState(ds *appsv1.DaemonSet) {
+	// Copy required labels
+	for key := range rbdDaemonsetLabels {
+		ds.Labels[key] = cephfsDaemonsetLabels[key]
+	}
+
+	// Update the demaon set with desired state
+	rbdDeamonSetSpec.DeepCopyInto(&ds.Spec)
+
+	// Update containers spec with desired state
+	for i := range ds.Spec.Template.Spec.Containers {
+		c := &ds.Spec.Template.Spec.Containers[i]
+		switch c.Name {
+		case templates.DriverRegistrar.Name:
+			c.Image = sidecarImages.ContainerImages.DriverRegistrarImageURL
+			c.Args = append(c.Args, fmt.Sprintf(
+				"--kubelet-registration-path=%s/plugins/%s/csi.sock",
+				templates.DefaultKubeletDirPath,
+				GetRBDDriverName(),
+			))
+
+		case templates.CSIAddonsContainer.Name:
+			c.Image = sidecarImages.ContainerImages.CSIADDONSImageURL
+			vol := utils.Find(c.VolumeMounts, func(vol *corev1.VolumeMount) bool {
+				return vol.Name == "socket-dir"
+			})
+			vol.Name = "plugin-dir"
```

Comment:
I think this panics due to above mentioned reason, we are not adding volumemount to the container and so `vol` will be nil here.

Reply:
Fixed now